sbomtool/pkg/generate/generate.go

```diff
@@ -0,0 +1,136 @@
+package generate
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/format/cyclonedxjson"
+	"github.com/anchore/syft/syft/format/spdxjson"
+	"github.com/anchore/syft/syft/sbom"
+)
+
+// closeFile properly closes a file and handles any close errors.
+// It logs close errors to stderr but doesn't return them to avoid
+// overriding more important errors from the main operation.
+func closeFile(f *os.File, operation string) {
+	if closeErr := f.Close(); closeErr != nil {
+		slog.Error("Failed to close file", "operation", operation, "error", closeErr)
+	}
+}
+
+// createSpdxSbom creates an SPDX SBOM file
+func createSpdxSbom(sbom sbom.SBOM, name string, outDir string) error {
+	cfg := spdxjson.DefaultEncoderConfig()
+	cfg.Pretty = false
+	cfg.Version = "2.3"
+
+	encoder, err := spdxjson.NewFormatEncoderWithConfig(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to create SPDX encoder: %w", err)
+	}
+
+	outputPath := filepath.Join(outDir, name+"-spdx.json")
+	f, err := os.OpenFile(outputPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return fmt.Errorf("failed to create SPDX output file: %w", err)
+	}
+	defer closeFile(f, "SPDX")
+
+	if err = encoder.Encode(f, sbom); err != nil {
+		return fmt.Errorf("failed to encode SPDX SBOM: %w", err)
+	}
+
+	return nil
+}
+
+// createCyclonedxSbom creates a CycloneDX SBOM file
+func createCyclonedxSbom(sbom sbom.SBOM, name string, outDir string) error {
+	cfg := cyclonedxjson.DefaultEncoderConfig()
+	cfg.Pretty = false
+	cfg.Version = "1.6"
```

Comment:
Does this mean `1.6.x` or `1.6` only. Because there is a 1.6.1 release of the CycloneDX specification.

https://github.com/CycloneDX/specification/releases

Reply:
> Functionally equivalent to CycloneDX 1.6.0
The spec overview also calls out just `1.6` as the latest.